feature/migration/qrcode/src/main/kotlin/app/k9mail/feature/migration/qrcode/domain/usecase/QrCodeResolutionFilter.kt

```diff
@@ -0,0 +1,30 @@
+package app.k9mail.feature.migration.qrcode.domain.usecase
+
+import android.util.Size
+import androidx.camera.core.resolutionselector.ResolutionFilter
+import kotlin.math.abs
+
+internal class QrCodeResolutionFilter : ResolutionFilter {
+    override fun filter(supportedSizes: List<Size>, rotationDegrees: Int): List<Size> {
+        var bestFit = supportedSizes.first()
+        var bestDistance = bestFit.distanceToTargetPixelCount()
+
+        for (size in supportedSizes.asSequence().drop(1)) {
+            val distance = size.distanceToTargetPixelCount()
+            if (distance <= bestDistance) {
+                bestDistance = distance
+                bestFit = size
+            }
+        }
+
+        return listOf(bestFit)
```

Comment:
```suggestion
        return supportedSizes.minByOrNull { it.distanceToTargetPixelCount() }
```

Reply:
Looking into this again, I found using `ResolutionStrategy` should be simpler and allows us to get rid of this whole class.

https://github.com/thunderbird/thunderbird-android/blob/bba75aed8df78569e33bc104a6d0315a321bcfdc/feature/migration/qrcode/src/main/kotlin/app/k9mail/feature/migration/qrcode/domain/usecase/QrCodeImageAnalysisProvider.kt#L22-L25